lib/plugins/aws/lib/naming.js

```diff
@@ -545,8 +545,17 @@ module.exports = {
   getLambdaAtEdgeInvokePermissionLogicalId(functionName) {
     return `${this.getLambdaLogicalId(functionName)}InvokePermission`;
   },
-  getCloudFrontOriginId(functionName, path) {
-    let originId = this.getNormalizedFunctionName(functionName);
+  getCloudFrontOriginId(originObj) {
+    const isS3Origin = originObj.S3OriginConfig !== undefined;
+    const domain = originObj.DomainName;
+    const path = originObj.OriginPath;
+
+    let originId = isS3Origin ? 's3' : 'custom';
+    const domainName = _.isString(domain)
+      ? domain
+      : this.normalizeNameToAlphaNumericOnly(JSON.stringify(domain));
```

Comment:
What is the use case for configuring `originObj.DomainName` as an object?

Reply:
When you are referencing to the cloudformation variables/functions eg:
```
    origin:
      DomainName:
        Fn::Join:
          - ''
          - - Ref: ApiGatewayRestApi
            - ".execute-api."
            - Ref: AWS::Region
            - ".amazonaws.com"